hetmech/hetmat.py

```diff
@@ -197,6 +224,71 @@ def permutations(self):
         return permutations
 
+    def permute_graph(self, num_new_permutations=None, namer=None, start_from=None,
+                      multiplier=10, excluded_pair_set=set(), seed=0, log=True):
+        """
+        Generate and save permutations of the HetMat adjacency matrices.
+
+        Parameters
+        ----------
+        num_new_permutations : int
+            The number of new, permuted HetMats to generate
+        namer : generator
+            Yields the names of new permutations. Cannot pass names of existing permutations
+        start_from : str
+            Name of permutation to use as starting point. For multiple permutations,
+            the first permutation starts from start_from, and future permutations
+            continue from the previous one.
+        multiplier : int
+            How many attempts to make when cross-swapping edges.
+        excluded_pair_set : set
+            Pairs of nodes which are not to be changed
+        seed : int
+            Random seed for generating new permutations
+        log : bool
+            When log=True, give information about permutation attempts
+        """
+        if namer is None:
+            # If no namer given, continue increasing names by one for new permutations
+            def namer():
+                i = 1
+                while True:
+                    yield str(i).zfill(3)
+                    i += 1
+            namer = namer()
+
+        logging_stats = list()
+        for _ in range(num_new_permutations):
+            permutation_name = next(namer)
+            name = f'{permutation_name}.hetmat/'
+            new_permutation_path = self.permutations_directory.joinpath(name)
+
+            new_hetmat = HetMat(new_permutation_path, initialize=True)
+            new_hetmat.is_permutation = True
+            new_hetmat.metagraph_path.symlink_to('../../metagraph.json')
+            new_hetmat.nodes_directory.rmdir()
+            new_hetmat.nodes_directory.symlink_to('../../nodes', target_is_directory=True)
+
+            if start_from is not None:
+                original_hetmat = self.permutations[start_from]
+            else:
+                original_hetmat = self
+
+            metaedges = list(self.metagraph.get_edges(exclude_inverts=True))
+            for metaedge in metaedges:
+                rows, cols, original_matrix = original_hetmat.metaedge_to_adjacency_matrix(
+                    metaedge, dense_threshold=1)
+                is_directed = metaedge.direction != 'both'
+                permuted_matrix, stats = _permute_matrix(
+                    original_matrix, directed=is_directed, multiplier=multiplier,
+                    excluded_pair_set=excluded_pair_set, seed=seed, log=log)
+                path = new_hetmat.get_edges_path(metaedge, file_format=None)
+                save_matrix(permuted_matrix, path)
+                logging_stats.append(stats)
+            start_from = permutation_name
```

Comment:
This does not seem neccessary at this point?

Reply:
If not, then we need to increment the seed after each iteration. Otherwise would we not be outputting multiple copies of the same permutation?

Reply:
Disregard this comment... I was misunderstanding